### PR TITLE
Add integration testing with geth

### PR DESCRIPTION
**Pull request by ed255**
Introduce a new crate `integration-tests` to do integration tests of the
rest of the crates with geth.  This crate contains a binary used to
generate data for the geth dev blockchain, and groups of integration
tests.  For now we only have the `rpc` test group.  See
`integration-tests/README.md` for more details.

Add a github action workflow that runs the integration test for the
bus-mapping rpc methods.  The action runs the `run.sh` integration test
script by steps to make it easier to see how long steps take, and to
quickly find which step fails in case of an error.

bus-mapping: Add `get_code_by_address` rpc method.

bus-mapping: Set the fields in the types returned by `get_proof` public,
and move them to `eth_types`.

Co-authored by: NoCtrlZ <phantomofrotten@gmail.com>

Resolve https://github.com/appliedzkp/zkevm-circuits/issues/206
Resolve https://github.com/appliedzkp/zkevm-circuits/issues/194

### CODE REVIEW

File: integration-tests/README.md [hunk 1]
```diff
@@ -0,0 +1,42 @@
+# Integration Tests
+
```
Comment (ChihChengLiang) [inline integration-tests/README.md:2]: Can add prerequisite of solc v0.8.0

I'm getting this error
```
$  ./run.sh 
...
thread 'main' panicked at 'solc compile error: Io(Os { code: 2, kind: NotFound, message: "No such file or directory" })', integration-tests/src/bin/gen_blockchain_data.rs:51:14
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Review by ChihChengLiang — APPROVED** [review]
Add a small suggestion on the README. Other than that LGTM. The script works fine on my device.

File: integration-tests/README.md (hunk 1 above)
Comment (ed255) [reply integration-tests/README.md:2]: I've added a `Requirements` section in the readme :)

File: integration-tests/run.sh [hunk 2]
```diff
@@ -0,0 +1,108 @@
+#!/bin/sh
+set -e
+
+ARG_DEFAULT_SUDO=
+ARG_DEFAULT_STEPS="setup gendata tests cleanup"
+ARG_DEFAULT_TESTS="rpc"
+# ARG_DEFAULT_TESTS="rpc bus-mapping"
```
Comment (CPerezz) [inline integration-tests/run.sh:7]: Leftover maybe?

**Review by CPerezz — APPROVED** [review]
AMAZING WORK @ed255 !!!!!

**Pull request merged**